### PR TITLE
[@jsreport/nodejs-client] added types for @jsreport/nodejs-client

### PR DESCRIPTION
**Pull request by reediculous456**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/jsreport__nodejs-client/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import JsReport = require('jsreport');
+import { ServerResponse } from 'http';
+
+declare namespace JsReportNodeJsClient {
+    interface ClientRenderResponse extends ServerResponse {
+        body(): Promise<Buffer>;
+    }
+
+    interface Client {
+        render(req: Partial<JsReport.Request>, options?: object): Promise<ClientRenderResponse>;
```
Comment (peterblazejewicz) [inline types/jsreport__nodejs-client/index.d.ts:15]:
```suggestion
        /** @async */
        render(req: Partial<JsReport.Request>, options?: object): Promise<ClientRenderResponse>;
```
https://jsdoc.app/tags-async.html#:~:text=The%20%40async%20tag%20indicates%20that%20a%20function%20is,tag%20is%20available%20in%20JSDoc%203.5.0%20and%20later.
`.d.ts` lacks ability to use `async` keyword

File: types/jsreport__nodejs-client/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    }
+}
+
+declare function CreateJsReportClient(url: string, username?: string, password?: string): JsReportNodeJsClient.Client;
+export = CreateJsReportClient;
```
Comment (peterblazejewicz) [inline types/jsreport__nodejs-client/index.d.ts:20]:
```suggestion
declare function createJsReportClient(url: string, username: string, password: string): JsReport.Client;
declare function createJsReportClient(url: string): JsReport.Client;

export = createJsReportClient;
```
I understand this are types for a aliased/fork package. But that method can be overloaded, as per usage. CamelCase is not a JS convention for functions that are not constructor type, and this is irrelevant to usage

File: types/jsreport__nodejs-client/jsreport__nodejs-client-tests.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import JsReport = require('jsreport');
+import Client = require('@jsreport/nodejs-client');
```
Comment (peterblazejewicz) [inline types/jsreport__nodejs-client/jsreport__nodejs-client-tests.ts:2]:
```suggestion
import client = require('@jsreport/nodejs-client');
```
https://github.com/jsreport/jsreport/blob/master/packages/nodejs-client/test/clientTest.js#L8

File: types/jsreport__nodejs-client/jsreport__nodejs-client-tests.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+(async () => {
+    const jsreport = JsReport();
+    await jsreport.init();
+
+    const client = Client('http://localhost:5488');
+    await client.render({
```
Comment (peterblazejewicz) [inline types/jsreport__nodejs-client/jsreport__nodejs-client-tests.ts:9]:
```suggestion
    await client('http://localhost:5488').render({
```

**Review by gabritto — APPROVED** [review]

**Comment by reediculous456** [comment]
Ready to merge

**Pull request merged**